balancer-js/src/modules/liquidity-managment/migrations.integrations.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { impersonateAccount, reset } from '@/test/lib/utils';
+import { expect } from 'chai';
+import { Vault__factory } from '@/contracts';
+import { BALANCER_NETWORK_CONFIG } from '@/lib/constants/config';
+import { JsonRpcProvider, JsonRpcSigner } from '@ethersproject/providers';
+import { ERC20 } from '@/modules/contracts/implementations/ERC20';
+import {
+  metaStable,
+  composableStable,
+  poolRepository,
+  gaugesRepository,
+} from './migrations/builder.spec-helpers';
+import { Migrations } from './migrations';
+
+const {
+  addresses: { contracts },
+} = BALANCER_NETWORK_CONFIG[1];
+const relayerAddress = contracts.relayerV4 as string;
+const provider = new JsonRpcProvider('http://127.0.0.1:8545');
+const vault = Vault__factory.connect(contracts.vault, provider);
+let signer: JsonRpcSigner;
+let address: string;
+
+const migrations = new Migrations(
+  relayerAddress,
+  poolRepository,
+  gaugesRepository,
+  provider
+);
+
+describe('Migrations', () => {
+  beforeEach(async () => {
+    await reset('https://rpc.ankr.com/eth', provider, 16820000);
+    signer = await impersonateAccount(address, provider);
+
+    // approve relayer
+    await vault
+      .connect(signer)
+      .setRelayerApproval(address, relayerAddress, true);
+  });
+
+  context('Metastable to Metastable', () => {
+    const from = metaStable;
+    const to = from;
+    const pool = from.address;
+
+    describe('bptHodler', () => {
+      before(() => {
+        address = '0x21ac89788d52070D23B8EaCEcBD3Dc544178DC60';
+      });
+
+      it('should execute a migration using exit / join', async () => {
+        const balance = await ERC20(pool, signer).balanceOf(address);
+
+        const txParams = await migrations.pool2pool(address, from.id, to.id);
+
+        await (await signer.sendTransaction(txParams)).wait();
+
+        const balanceAfter = await ERC20(pool, signer).balanceOf(address);
+        const diff = Number(balance.sub(balanceAfter));
+
+        expect(diff).to.be.lt(1e18);
```

Comment:
Rejected transactions should throw and this dust idea was best I could come up with exactly for the reason it's using the same pool. Once new pools are deployed I'll update the tests as well.

Reply:
Isn't there any pair of pools you're able to test migration? I understand having from/to being the same pool as initial tests to get things going, but it would be nice to have more realistic test scenarios before merging the PR.

Reply:
I found two pools with the same token pairs and added a test for it.